// langchain/dist/output_parsers/regex.js
import { BaseOutputParser } from "./base.js";
/**
 * Class to parse the output of an LLM call into a dictionary.
 * @augments BaseOutputParser
 */
export class RegexParser extends BaseOutputParser {
    constructor(regex, outputKeys, defaultOutputKey) {
        super();
        Object.defineProperty(this, "regex", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "outputKeys", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        Object.defineProperty(this, "defaultOutputKey", {
            enumerable: true,
            configurable: true,
            writable: true,
            value: void 0
        });
        this.regex = regex;
        this.outputKeys = outputKeys;
        this.defaultOutputKey = defaultOutputKey;
    }
    _type() {
        return "regex_parser";
    }
    parse(text) {
        const match = text.match(this.regex);
        if (match) {
            return this.outputKeys.reduce((acc, key, index) => {
                acc[key] = match[index + 1];
                return acc;
            }, {});
        }
        if (this.defaultOutputKey === undefined) {
            throw new Error(`Could not parse output: ${text}`);
        }
        return this.outputKeys.reduce((acc, key) => {
            acc[key] = key === this.defaultOutputKey ? text : "";
            return acc;
        }, {});
    }
    getFormatInstructions() {
        return `Your response should match the following regex: /${this.regex}/`;
    }
    serialize() {
        return {
            _type: "regex_parser",
            regex: typeof this.regex === "string" ? this.regex : this.regex.source,
            output_keys: this.outputKeys,
            default_output_key: this.defaultOutputKey,
        };
    }
    static async deserialize(data) {
        return new RegexParser(data.regex, data.output_keys, data.default_output_key);
    }
}
